internal/staging/primitives/uint256/uint256_test.go

```diff
@@ -0,0 +1,4064 @@
+// Copyright (c) 2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package uint256
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"reflect"
+	"testing"
+	"time"
+)
+
+// hexToBytes converts the passed hex string into bytes and will panic if there
+// is an error.  This is only provided for the hard-coded constants so errors in
+// the source code can be detected. It will only (and must only) be called with
+// hard-coded values.
+func hexToBytes(s string) []byte {
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		panic("invalid hex in source file: " + s)
+	}
+	return b
+}
+
+// hexToUint256 converts the passed hex string into a Uint256 and will panic if
+// there is an error.  This is only provided for the hard-coded constants so
+// errors in the source code can be detected. It will only (and must only) be
+// called with hard-coded values.
+func hexToUint256(s string) *Uint256 {
+	if len(s)%2 != 0 {
+		s = "0" + s
+	}
+	b := hexToBytes(s)
+	if len(b) > 32 {
+		panic("hex in source file overflows mod 2^256: " + s)
+	}
+	return new(Uint256).SetByteSlice(b)
+}
+
+// randBigIntAndUint256 returns a big integer and a uint256 both created from
+// the same random value generated by the passed rng.
+func randBigIntAndUint256(t *testing.T, rng *rand.Rand) (*big.Int, *Uint256) {
```

Comment:
It might be worth using `t.Helper()` here for cleaner failure output. 

```suggestion
func randBigIntAndUint256(t *testing.T, rng *rand.Rand) (*big.Int, *Uint256) {
    t.Helper()
```

Reply:
Agreed.  I usually do that, but forgot it here.  Nice catch.